### ISSUE TITLE
Rancher Desktop executing wrong tar.exe internally

### ISSUE THREAD

**Issue by Technophobe01** [rancher-sandbox/rancher-desktop#1807]
### Actual Behavior

I tried to install Rancher Desktop on a Windows 11 Machine, the install failed 

Error:
Error: UNKNOWN: unknown error, open '\\wsl$\rancher-desktop-data\etc\hosts'
Recent logfile lines:
2022-03-14T17:09:54.415Z: Running command wsl --distribution rancher-desktop --exec wslpath -a -u C:\Users\pkjar\AppData\Local\Temp\rd-distro-HKdjWz\distro.tar...
2022-03-14T17:09:54.581Z: Running command wsl --distribution rancher-desktop --exec tar -cf /mnt/c/Users/pkjar/AppData/Local/Temp/rd-distro-HKdjWz/distro.tar -C / /bin/busybox /bin/mount /bin/sh /lib /etc/passwd /etc/rancher /var/lib...
2022-03-14T17:09:55.538Z: Error registering data distribution: Error: tar.exe exited with code 2
2022-03-14T17:09:55.538Z: Running command wsl --unregister rancher-desktop-data...
2022-03-14T17:09:55.606Z: WSL failed to execute wsl.exe --unregister rancher-desktop-data: Error: wsl.exe exited with code 4294967295
2022-03-14T17:09:55.607Z: Error setting up data distribution: Error: wsl.exe exited with code 4294967295
    at ChildProcess.<anonymous> (C:\Users\pkjar\AppData\Local\Programs\Rancher Desktop\resources\app.asar\dist\app\background.js:1:8692)
    at ChildProcess.emit (node:events:394:28)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)

When I tried to uninstall Rancher Desktop the uninstall failed - 

Error "Rancher Desktop cannot be closed. Please close is manually and Click Retry to continue"

Windows 11 Pro
Version 21H2
OS BUild: 22000.556
Intel(R) Xeon(R) CPU E3-1505M v5 @ 2.80GHz   2.81 GHz
RAM 64.0 GB (63.8 GB usable)

![Install Fails](https://user-images.githubusercontent.com/749230/158226435-82b23a5e-898a-4f4c-9bb5-0a8a22922fd3.png)
![Rancher Uninstall](https://user-images.githubusercontent.com/749230/158226455-900ce412-7979-4a21-931f-8123cd572462.png)

### Steps to Reproduce

Download installer and run it on Windows 11, - install fails.
launch uninstall and get prompted to shutdown rancher

### Result

Error: On Install

Error: UNKNOWN: unknown error, open '\\wsl$\rancher-desktop-data\etc\hosts'
Recent logfile lines:
2022-03-14T17:09:54.415Z: Running command wsl --distribution rancher-desktop --exec wslpath -a -u C:\Users\pkjar\AppData\Local\Temp\rd-distro-HKdjWz\distro.tar...
2022-03-14T17:09:54.581Z: Running command wsl --distribution rancher-desktop --exec tar -cf /mnt/c/Users/pkjar/AppData/Local/Temp/rd-distro-HKdjWz/distro.tar -C / /bin/busybox /bin/mount /bin/sh /lib /etc/passwd /etc/rancher /var/lib...
2022-03-14T17:09:55.538Z: Error registering data distribution: Error: tar.exe exited with code 2
2022-03-14T17:09:55.538Z: Running command wsl --unregister rancher-desktop-data...
2022-03-14T17:09:55.606Z: WSL failed to execute wsl.exe --unregister rancher-desktop-data: Error: wsl.exe exited with code 4294967295
2022-03-14T17:09:55.607Z: Error setting up data distribution: Error: wsl.exe exited with code 4294967295
    at ChildProcess.<anonymous> (C:\Users\pkjar\AppData\Local\Programs\Rancher Desktop\resources\app.asar\dist\app\background.js:1:8692)
    at ChildProcess.emit (node:events:394:28)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)

Error on Uninstall

When I tried to uninstall Rancher Desktop the uninstall failed - 

Error "Rancher Desktop cannot be closed. Please close is manually and Click Retry to continue
![Install Fails](https://user-images.githubusercontent.com/749230/158226841-c382afe5-ee01-4e48-a9d4-9793f350d149.png)
![Uploading Install Fails.png…]()
![Rancher Uninstall](https://user-images.githubusercontent.com/749230/158226848-8ee247ec-a1bc-484c-b8d7-2cd7ac433df3.png)
"

### Expected Behavior

Clean install and uninstall

### Rancher Desktop Version

1.1.1

### Rancher Desktop K8s Version

1.22.7

### Which container runtime are you using?

containerd (nerdctl)

### What operating system are you using?

Windows

### Operating System / Build Version

Windows 11 Pro Version 21H2 OS BUild: 22000.556

### What CPU architecture are you using?

x64

### Windows User Only

NordVPn was installed

**Comment by gravufo** [comment]
I actually have the same issue! I wasn't sure what was going on, but at least I'm not the only one lol

Hope someone can help pinpoint and fix the issue! I'm willing to test out new versions if necessary.

**Comment by Technophobe01** [comment]
I am wondering if the VPN had anything to do with the failure. Do you have a VPN installed? 

> wsl -l -v
```
  NAME                                STATE           VERSION
* Ubuntu20.04LTS             Stopped         2
  openSUSE-Leap-15.3      Stopped         2
  rancher-desktop              Stopped         2
```

**Comment by gravufo** [comment]
No, I don't have a VPN currently running, although I do have OpenVPN installed on the machine.

**Comment by bkmeneguello** [comment]
Looks like [the error reported by OP](https://github.com/rancher-sandbox/rancher-desktop/issues/1807#issue-1168704677) is due to the fact [tar.exe](https://github.com/rancher-sandbox/rancher-desktop/blob/5ce0209aac1659c46805b657cb7f8c99c6f1fa41/src/k8s-engine/wsl.ts#L506) is supposed to be Window's default, but in some cases (like mine) it was replaced by `busybox` tar.

**Comment by jboyd01** [comment]
I do have cygwin with tar which has precedence over the Windows version in my PATH...  I can't reproduce a syntax error though using cygwin tar vs windows tar using the command format of   `tar -cf  <some file> -C <some directory> file1 file2 file3` unless I specify non existent files.  I'm uncertain of the context where tar is running though to pick up these paths of the files to tar up:   `/bin/busybox `, `/bin/mount`, etc don't exist under standard windows or cygwin context.

**Comment by Technophobe01** [comment]
@jboyd01 This fixed the issue for me. 
Steps to resolve:
1. Open Windows Powershell
2. > Get-Command tar 
3. Result "C:\Rtools\bin"
4. > sysdm.cpl
5. Select the Advanced tab
6. Select Environment Variables (Bottom of window)
7. Select path (bottom window)
8. Move "C:\Rtools\bin" below "C:\WINDOWS\system32"
9. > Get-Command tar
10. Result "C:\WINDOWS\system32\tar.exe"
11. Open Rancher Desktop and factory reset
12. Result: Rancher is now runing.

**Comment by jboyd01** [comment]
I had to go back to windows 10 for other issues but I agree, changing the path works around the issue.  Thanks @Technophobe01

**Comment by micahmo** [comment]
I had this problem as well; for me it was because `C:\Program Files\Git\usr\bin` was higher than `C:\Windows\system32` in my `PATH`. Since Rancher Desktop clearly wants to use the Windows version, could they look for the whole path?

**Comment by ron-stauffer** [comment]
Turn out to be that I had cygwin installed and it was using its tar exe. Hint found here: https://github.com/rancher-sandbox/rancher-desktop/issues/1807

**Comment by gaktive** [comment]
@sunilarjun since this issue pops up a lot, do you think we should add a troubleshooting step or common problem thing for this error somewhere?

**Comment by sunilarjun** [comment]
@gaktive I think that's a good idea! I can add a troubleshooting tip in this [section](https://docs.rancherdesktop.io/troubleshooting-tips#installation) of the docs with some steps described here on changing the path to resolve the installation issue.

**Comment by gaktive** [comment]
Thanks @sunilarjun! -- hold off a bit since there's a chance we could fix this but we'd have to discuss and schedule that first. Sounds like a quick one but it's juggling priorities.

**Comment by mook-as** [comment]
This was fixed by #6691. (It's not shipped yet.)
